File: src/types/Theme.ts
import type { MediaState, Schedule } from "./Schedule";

export interface ThemeAvailability {
  id: string;
  availableLanguages: string[];
}

export interface Theme {
  id?: string | null;
  mediaState: MediaState;
  schedule?: Schedule | null;
}

export function getThemeId(theme: Theme): string | null {
  return theme.id ?? theme.schedule?.id ?? theme.schedule?.scheduleId ?? null;
}

export function getThemeScheduleId(theme: Theme): string | null {
  return theme.schedule?.id ?? theme.schedule?.scheduleId ?? null;
}

File: src-tauri/src/models/theme.rs
use crate::models::schedule::Schedule;
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ThemeAvailability {
    pub id: String,
    pub available_languages: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct Theme {
    #[serde(skip_serializing_if = "Option::is_none")]
    pub id: Option<String>,
    pub media_state: crate::models::schedule::MediaState,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub schedule: Option<Schedule>,
}
